be/src/service/service_be/lake_service.cpp

```diff
@@ -257,7 +257,62 @@ void LakeServiceImpl::publish_log_version(::google::protobuf::RpcController* con
         auto st = thread_pool->submit_func(task);
         if (!st.ok()) {
             g_publish_version_failed_tasks << 1;
-            LOG(WARNING) << "Fail to submit publish log version task: " << st;
+            LOG(WARNING) << "Fail to submit publish log version task, tablet_id: " << tablet_id
+                         << ", txn_id: " << txn_id << ", version: " << version << ", error: " << st;
+            std::lock_guard l(response_mtx);
+            response->add_failed_tablets(tablet_id);
+            latch.count_down();
+        }
+    }
+
+    latch.wait();
+}
+
+void LakeServiceImpl::publish_log_version_batch(::google::protobuf::RpcController* controller,
```

Comment:
The logic in the first method (publish_log_version) can be simplified by moving the common variable assignments outside the for loop.
The second method (publish_log_version_batch) seems to handle batch processing but has similar logic to the first method.
Consider refactoring the code to eliminate code duplication.

this suggestion seems reasonable.